packages/turf-line-offset/index.js

```diff
@@ -0,0 +1,70 @@
+var getCoords = require('@turf/invariant').getCoords;
+var coordEach = require('@turf/meta').coordEach;
+var intersection = require('intersection');
+var helpers = require('@turf/helpers');
+var lineString = helpers.lineString;
+var distanceToDegrees = helpers.distanceToDegrees;
+
+/**
+ * Takes a {@link LineString|line} and returns a {@link LineString|line} at offset by the specified distance.
+ *
+ * @name lineOffset
+ * @param {Feature<LineString>|Geometry<LineString>} line input line
+ * @param {number} offset distance to offset the line
+ * @param {string} [units=kilometers] can be degrees, radians, miles, kilometers, inches, yards, meters
+ * @returns {Feature<LineString>} Line offset from the input line
+ * @example
+ * var line = {
+ *   "type": "Feature",
+ *   "properties": {},
+ *   "geometry": {
+ *     "type": "LineString",
+ *     "coordinates": [[-83, 30], [-84, 36], [-78, 41]]
+ *   }
+ * };
+ *
+ * var offsetLine = turf.lineOffset(line, 2, 'miles');
+ *
+ * //addToMap
+ * var addToMap = [offsetLine, line]
+ */
+module.exports = function (line, offset, units) {
+    var segments = [];
+    var offsetDegrees = distanceToDegrees(offset, units);
+    var coords = getCoords(line);
+
+    coordEach(line, function (currentCoords, currentIndex) {
+        if (currentIndex !== coords.length - 1) {
+            var outCoords = processSegment(currentCoords, coords[currentIndex + 1], offsetDegrees);
+            segments.push(outCoords);
```

Comment:
Will do :) I'll have a look at it more tomorrow, it's a good starting point so far!
+1